drivers/mm/mm_drv_intel_adsp_mtl_tlb.c

```diff
@@ -187,8 +187,6 @@ int sys_mm_drv_map_page(void *virt, uintptr_t phys, uint32_t flags)
 	uintptr_t pa = POINTER_TO_UINT(sys_cache_cached_ptr_get(UINT_TO_POINTER(phys)));
 	uintptr_t va = POINTER_TO_UINT(sys_cache_cached_ptr_get(virt));
 
-	ARG_UNUSED(flags);
```

Comment:
if `CONFIG_MMU` is undefined `flags` is only used as an argument to `flags_to_tlb_perms()`, which is a `static` function, defined in this file. Strangely, it has `#if defined(CONFIG_SOC_SERIES_INTEL_ADSP_ACE)` in it, can this file even be compiled (let alone used) if `CONFIG_SOC_SERIES_INTEL_ADSP_ACE` is undefined? In *that* case, that function would probably be inlined by the compiler, then `flags` would indeed become unused. But I guess that `#if` can be removed from that function

Reply:
This macro is used only to prevent the compiler from issuing a warning about an unused parameter. Since it is always passed as a parameter to a function, its use is unnecessary. It does not matter whether the compiler decides to inline the function. But in fact, this macro should be used in the `flags_to_tlb_perms` function so that the compiler does not generate a warning if the `CONFIG_SOC_SERIES_INTEL_ADSP_ACE` option is not enabled.